ansible_collections/arista/avd/roles/eos_designs/python_modules/overlay/router_path_selection.py

```diff
@@ -44,44 +45,39 @@ def _get_path_groups(self) -> list:
         """
         Generate the required path-groups locally
         """
-        # TODO - get this once WAN interface are available
-        # TODO - this function will need to handle Crossconnection of public path_groups
-        #
-        local_path_groups = []
-        # The value will be set based on the WAN interfaces configuration
-        # local_path_groups = self.shared_utils.path_groups
-
-        if not local_path_groups:
-            return []
-
         path_groups = []
 
         # TODO - need to have default value in one place only -> maybe facts / shared_utils ?
-        ipsec_profile_name = get(self._hostvars, "wan_ipsec_profiles.control_plane.profile_name", required=True)
-
-        for carrier in local_path_groups:
-            path_groups.append(
-                {
-                    "name": carrier.get("name"),
-                    "id": self._get_carrier_id(carrier),
-                    "ipsec_profile": ipsec_profile_name,  # TODO - disable on a per carrier basis
-                    "local_interfaces": self._get_local_interfaces(carrier),
-                    "dynamic_peers": self._get_dynamic_peers(),
-                    "static_peers": self._get_static_peers(carrier),
-                }
-            )
+        ipsec_profile_name = get(self._hostvars, "wan_ipsec_profiles.control_plane.profile_name", default="CP-PROFILE")
+
+        for path_group in self.shared_utils.wan_local_path_groups:
+            pg_name = path_group.get("name")
+
+            path_group_data = {
+                "name": pg_name,
+                "id": self._get_path_group_id(pg_name, path_group.get("id")),
+                "local_interfaces": self._get_local_interfaces_for_path_group(pg_name),
+                "dynamic_peers": self._get_dynamic_peers(),
+                "static_peers": self._get_static_peers_for_path_group(pg_name),
+            }
+
+            if path_group.get("ipsec", True):
+                path_group_data["ipsec_profile"] = ipsec_profile_name
+
+            path_groups.append(path_group_data)
 
         if self.shared_utils.cv_pathfinder_role:
             pass
             # implement LAN_HA here
 
-        return path_groups
+        return natural_sort(path_groups, "name")
```

Comment:
I think this sort is what causes the CI failures. It was not there before, so you are reordering. You need to regenerate artifacts.